src/commands/synthetics/utils.ts

```diff
@@ -475,40 +521,26 @@ export const getTestsToTrigger = async (
     )
   }
 
-  const tests = await Promise.all(
-    triggerConfigs.map(async ({config, id, suite}) => {
-      let test: Test | undefined
-      id = PUBLIC_ID_REGEX.test(id) ? id : id.substr(id.lastIndexOf('/') + 1)
-      try {
-        test = {
-          ...(await api.getTest(id)),
-          suite,
-        }
-      } catch (error) {
-        if (isNotFoundError(error)) {
-          summary.testsNotFound.add(id)
-          const errorMessage = formatBackendErrors(error)
-          errorMessages.push(`[${chalk.bold.dim(id)}] ${chalk.yellow.bold('Test not found')}: ${errorMessage}`)
-
-          return
-        }
+  const summary = createSummary()
+  const testsAndConfigsOverride = await Promise.all(
+    triggerConfigs.map((triggerConfig) => getTestAndOverrideConfig(api, triggerConfig, reporter, summary))
+  )
 
-        throw error
-      }
+  const overriddenTestsToTrigger: TestPayload[] = []
+  const waitedTests: Test[] = []
+  testsAndConfigsOverride.forEach(({test, errorMessage, overriddenConfig}) => {
+    if (errorMessage) {
+      errorMessages.push(errorMessage)
+    }
 
-      const overriddenConfig = handleConfig(test, id, reporter, config)
+    if (overriddenConfig) {
```

Comment:
Is there a reason why we check if `overriddenConfig` is truthy? (whereas before we didn't care)

Reply:
Previously everything was done in the same function, and thee push was done only if we successfully get the config https://github.com/DataDog/datadog-ci/pull/609/files#diff-741a985551714cec0c6f7a6423d704525619e8f57db6c4f7c685d95d88a8d0cfL499. Now, now we return the overwritten config instead of directly adding it to the array.